test/dsets.c

```diff
@@ -5899,6 +5899,14 @@ test_floattypes(hid_t file)
         if ((datatype = H5Tcopy(H5T_NATIVE_LDOUBLE)) < 0)
             goto error;
 
+            /* Skip creating a custom floating-point type when long double
+             * is the IBM long double. The library detects different formats
+             * for the type on big-endian vs. little-endian systems and will
+             * cause this to fail on little-endian systems while passing on
+             * big-endian systems. The library needs proper support for the
+             * IBM long double type before we can test this.
+             */
+#if LDBL_MANT_DIG != 106
```

Comment:
I plan to eventually fix the library's configure checks and convert this to a more meaningful #ifdef, but for now this will do for IBM long double detection.